src/components/AppComponent.native.tsx

```diff
@@ -67,6 +69,10 @@ const AppComponent: React.FC = () => {
     if (showSidebar) return openDrawer()
   }, [showSidebar])
 
+  useEffect(() => {
+    setIsScrollEnabled(multiGestureRef.current?.isGestureActive || false)
+  }, [multiGestureRef])
```

Comment:
This mutable ref dependency will itself not re-run this effect. It is working because the `showAlert` is causing the `AppComponent` to re-render after every gesture, which is causing effect to work. Ideally, we would want to prevent the whole `AppComponent` to re-render when `alert` state changes.

Since we want a reactive state that syncs with the current status of gesture, `ref` is not an ideal option here. Instead, I think we can attach custom callbacks to the `MultiGesture` instead.

```javascript

/*** State ***/

const [isGestureActive, setIsGestureActive] = useState(false)

/*** Component ***/

  <MultiGesture
      onGesture={(...args) => {
        setIsGestureActive(true)
        handleGestureSegment(...args)
      }}
      onEnd={(...args) => {
        setIsGestureActive(false)
        handleGestureEnd(...args)
      }}
      shouldCancelGesture={shouldCancelGesture}
      onCancel={() => {
        setIsGestureActive(false)
        handleGestureCancel()
      }}
    >
      <ScrollView scrollEnabled={!isGestureActive}/>
    </MultiGesture>
```

Reply:
I believe this can also work. Also, this would help us prevent the dependency on the `showAlert ` selector changes which as you mentioned causes rerenders.